SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorGenerateSnapshot.cpp

```diff
@@ -281,31 +279,50 @@ bool CreateStartupActor(Worker_SnapshotOutputStream* OutputStream, AActor* Actor
 	check(ActorInfo);
 
 	WriteAclMap ComponentWriteAcl;
+
 	ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, UnrealServerPermission);
 	ComponentWriteAcl.Add(SpatialConstants::ROTATION_COMPONENT_ID, UnrealServerPermission);
-	ComponentWriteAcl.Add(ActorInfo->SingleClientComponent, UnrealServerPermission);
-	ComponentWriteAcl.Add(ActorInfo->MultiClientComponent, UnrealServerPermission);
-	ComponentWriteAcl.Add(ActorInfo->HandoverComponent, UnrealServerPermission);
-	// No write attribute for RPC_Client since a Startup Actor will have no owner on level start
-	ComponentWriteAcl.Add(ActorInfo->RPCComponents[RPC_Server], UnrealServerPermission);
-	ComponentWriteAcl.Add(ActorInfo->RPCComponents[RPC_CrossServer], UnrealServerPermission);
-	ComponentWriteAcl.Add(ActorInfo->RPCComponents[RPC_NetMulticast], UnrealServerPermission);
-
-	for (UClass* SubobjectClass : ActorInfo->SubobjectClasses)
+
+	ForAllSchemaComponentTypes([&](EComponentType Type) {
+		Worker_ComponentId ComponentId = ActorInfo->SchemaComponents[Type];
```

Comment:
Any chance we could combine this logic with that in CreateEntity?

Reply:
For sure, but I think that should be a different PR.